### DISCUSSION TITLE
Github Page support cleanUrls ?

### DISCUSSION THREAD

**Discussion by hf-xz** [vuejs/vitepress#3020] [Show and tell]
Following [vitepress guide](https://vitepress.dev/guide/routing#generating-clean-url) and [vitepress reference](https://vitepress.dev/reference/site-config#cleanurls), I set `cleanUrls` to true in both `vercel.json` and `config.mts` to generate clean url for my vercel deploy.

According to the guide, the Github Page doesn't support:

> If, however, you cannot configure your server with such support (e.g. GitHub pages)

However, after set `cleanUrls: true` in my `config.mts`, my Github Page also generate clean  url without any other configuration. Does this mean Github Page actually support cleanUrls? Or it's only my own case? *(I'm using github.io)*

My repository:

- https://github.com/hf-xz/hf-xz.github.io

you can check my directory structure here.

My deploy:

- Vercel: https://hfxz.link
- Github Page: https://hf-xz.github.io

you can see, both have clean url.

If the title is true, we should update the guide to tell others about this feature.

**Comment by brc-dd** [comment]
Ah, yeah it seems to work. We can update the docs. I don't think it used to work earlier.
